spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -666,18 +746,15 @@ function chanUpgradeOpen(
     proofChannel: CommitmentProof,
     proofHeight: Height,
 ) {
-    // if packet commitments are not empty then abort the transaction
-    abortTransactionUnless(pendingInflightPackets(portIdentifier, channelIdentifier))
-
-    // currentChannel must be in TRYUPGRADE or ACKUPGRADE and have completed flushing
+    // currentChannel must have completed flushing
     currentChannel = provableStore.get(channelPath(portIdentifier, channelIdentifier))
-    abortTransactionUnless(currentChannel.state == TRYUPGRADE || currentChannel.state == ACKUPGRADE)
-    abortTransactionUnless(currentChannel.flushStatus == FLUSHCOMPLETE)
+    abortTransactionUnless(currentChannel.state == FLUSHCOMPLETE)
 
+    // counterparty upgrade must not have passed on our chain
     connection = getConnection(currentChannel.connectionIdentifier)
     counterpartyHops = getCounterpartyHops(connection)
 
-    // counterparty must be in OPEN, TRYUPGRADE, ACKUPGRADE state
+    // counterparty must be in OPEN or FLUSHCOMPLETE state
     if counterpartyChannelState == OPEN {
         // get upgrade since counterparty should have upgraded to these parameters
         upgrade = provableStore.get(channelUpgradePath(portIdentifier, channelIdentifier))
```

Comment:
connectionHops in this one is incorrect. Should reference the counterparty of `upgrade.fields.connectionHops`. Could use counterparty upgrade if stored

Reply:
There is a [separate issue](https://github.com/cosmos/ibc/issues/1004) for this. I will fix it in a separate PR.